index.d.ts

```diff
@@ -99,7 +99,7 @@ export interface CancelStatic {
 }
 
 export interface Cancel {
-  message: string;
+  message: string | undefined;
```

Comment:
```suggestion
  message?: string;
```

Reply:
If I'm not wrong, the [implementation](https://github.com/NoriSte/axios/blob/master/lib/cancel/Cancel.js#L10) looks different, the `message` key is always present, but it could be `undefined`. My proposal reflects that while yours means that `message` could not exist. [This CodeSandbox](https://codesandbox.io/s/checking-promise-completion-on-axios-cancellation-lbl5j?file=/src/index.js:674-691) demonstrates it.

Reply:
I wrote this suggestion because I haven't found any `undefined` word in the file (even for CancelStatic & Canceler) so I thought it would be consistent. But your words sound very logical to me (thanks for the detailed explanation btw) so its up to you.